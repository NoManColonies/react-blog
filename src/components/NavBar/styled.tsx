import styled from 'styled-components'

interface ContainerPropType {
  isScroll: boolean;
}

export const Container = styled.div`
  width: 100%;
  position: ${(props: ContainerPropType): string =>
    props.isScroll ? 'fixed' : 'static'};
  z-index: 100;
  background-color: ${(props: ContainerPropType): string =>
    props.isScroll ? '#fff' : 'transparent'};
  transition: background-color 500ms;
`

export const Wrapper = styled.div`
  display: flex;
  flex: 1;
  padding: 2rem;
`

export const Logo = styled.div``

export const ActionContainer = styled.ul`
  display: flex;
  flex: 1;
  justify-content: flex-end;
`

export const Action = styled.li`
  padding: 0 1rem;
  color: #000;
  text-decoration: none;
`
