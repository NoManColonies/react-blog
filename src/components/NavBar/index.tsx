import React, { ReactElement } from 'react'
import useScroll from '../../utils/useScroll'
import { Container, Wrapper, ActionContainer, Action, Logo } from './styled'
import { Link } from 'react-router-dom'

function NavBar(): ReactElement {
  const isScroll = useScroll({})

  return (
    <Container isScroll={isScroll}>
      <Wrapper>
        <Logo>Logo</Logo>
        <ActionContainer>
          <Action>
            <Link to='/'>Home</Link>
          </Action>
          <Action>
            <Link to='/about'>About</Link>
          </Action>
          <Action>
            <Link to='/contact'>Contact</Link>
          </Action>
          <Action>
            <Link to='/story'>Story</Link>
          </Action>
        </ActionContainer>
      </Wrapper>
    </Container>
  )
}

export default NavBar
