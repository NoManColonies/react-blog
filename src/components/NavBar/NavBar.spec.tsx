/* global describe test expect */

import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { render } from '@testing-library/react'
import NavBar from '.'

describe('NavBar Component', () => {
  test('should contain logo text', () => {
    const { getByText } = render(
      <MemoryRouter>
        <NavBar></NavBar>
      </MemoryRouter>
    )

    expect(getByText(/logo/gi)).toBeInTheDocument()
  })

  test('should have link navigate to home, about, contact, and story', () => {
    const { getByText } = render(
      <MemoryRouter>
        <NavBar></NavBar>
      </MemoryRouter>
    )

    expect(getByText(/home/gi)).toBeInTheDocument()
    expect(getByText(/home/gi).closest('a')).toHaveAttribute('href', '/')

    expect(getByText(/about/gi)).toBeInTheDocument()
    expect(getByText(/about/gi).closest('a')).toHaveAttribute('href', '/about')

    expect(getByText(/contact/gi)).toBeInTheDocument()
    expect(getByText(/contact/gi).closest('a')).toHaveAttribute(
      'href',
      '/contact'
    )

    expect(getByText(/story/gi)).toBeInTheDocument()
    expect(getByText(/story/gi).closest('a')).toHaveAttribute('href', '/story')
  })

  test('should render layout correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <NavBar></NavBar>
      </MemoryRouter>
    )

    expect(container).toMatchSnapshot()
  })
})
