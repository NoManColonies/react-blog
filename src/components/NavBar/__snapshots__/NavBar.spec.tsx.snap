// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component should render layout correctly 1`] = `
<div>
  <div
    class="sc-bdnylx kzFGVQ"
  >
    <div
      class="sc-gtssRu gwgwje"
    >
      <div
        class="sc-dlnjPT"
      >
        Logo
      </div>
      <ul
        class="sc-hKFyIo ljvqsi"
      >
        <li
          class="sc-eCApGN heleiC"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="sc-eCApGN heleiC"
        >
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="sc-eCApGN heleiC"
        >
          <a
            href="/contact"
          >
            Contact
          </a>
        </li>
        <li
          class="sc-eCApGN heleiC"
        >
          <a
            href="/story"
          >
            Story
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
